lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -350,7 +350,78 @@ def disable(self) -> OpenSearchPluginConfig:
             config_entries_to_add={"knn.plugin.enabled": False},
         )
 
+    @property
+    def dependencies(self) -> Optional[List[str]]:
+        """Returns a list of plugin name dependencies."""
+        return ["opensearch-ml"]
```

Comment:
Could you confirm that this is needed? a very brief research suggested they're separate

Reply:
Actually, confirmed as you've said, that is not needed. The `train` API is actually different